trace-obfuscation/src/sql.rs

```diff
@@ -0,0 +1,1385 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0. This product includes software
+// developed at Datadog (https://www.datadoghq.com/). Copyright 2023-Present
+// Datadog, Inc.
+
+use crate::{
+    obfuscation_config::ObfuscationConfig,
+    sql_tokenizer::{SqlTokenizer, SqlTokenizerScanResult, TokenKind},
+};
+
+const QUESTION_MARK: char = '?';
+const QUESTION_MARK_STR: &str = "?";
+
+pub fn obfuscate_sql_string(s: &str, config: &ObfuscationConfig) -> AttemptSqlObfuscationResult {
+    let use_literal_escapes = false;
+    let mut tokenizer = SqlTokenizer::new(s, config.sql_literal_escapes);
+    let result = attempt_sql_obfuscation(tokenizer, config.sql_replace_digits);
+    if result.error.is_none() || !result.seen_escape {
+        return result;
+    }
+
+    tokenizer = SqlTokenizer::new(s, !use_literal_escapes);
```

Comment:
if obfuscation failed and we saw an escape character, try treating backslashes as character rather than escapes.

Reply:
We're going to want to store this in a global mutex eventually, so that future obfuscation attempts start with the correct literal escape config